File: boj/b_11052.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

n = int(input())
cards = [0] + list(map(int, input().rstrip().split()))
table = [0] * (n + 1)
for i in range(1, n + 1):
    table[i] = cards[i]
    for j in range(1, i + 1):
        table[i] = max(table[i], table[j] + table[i - j])
        if i % j == 0:
            table[i] = max(table[i], cards[j] * (i // j))
print(table[n])

File: boj/b_11052.js
function solution(n, cards) {
  table = Array(n + 1).fill(0);
  for (let i = 1; i <= n; i += 1) {
    table[i] = cards[i - 1];
    for (let j = 1; j <= i; j += 1) {
      table[i] = Math.max(table[i], table[j] + table[i - j]);
      if (i % j === 0) {
        table[i] = Math.max(table[i], cards[j - 1] * parseInt(i / j));
      }
    }
  }
  return table[n];
}
console.log(solution(10, [1, 100, 160, 1, 1, 1, 1, 1, 1, 1]));
